### PR TITLE
lib `git.rs`: remove workaround for a now-fixed libgit2 bug

### PR DESCRIPTION
**Pull request by ilyagr**
https://github.com/libgit2/libgit2/issues/3178 is now fixed.

### CODE REVIEW

File: lib/src/git.rs [hunk 1]
```diff
@@ -1275,50 +1275,30 @@ pub fn push_updates(
...
-                &temp_ref_name,
-                git2::Oid::from_bytes(new_target.as_bytes()).unwrap(),
-                true,
-                "temporary reference for git push",
-            )?);
             refspecs.push(format!(
                 "{}{}:{}",
                 (if update.force { "+" } else { "" }),
-                temp_ref_name,
+                git2::Oid::from_bytes(new_target.as_bytes()).unwrap(),
```
Comment (yuja) [inline lib/src/git.rs:1286]: nit: `new_target.hex()` also works

File: lib/src/git.rs [hunk 2]
```diff
@@ -1275,50 +1275,30 @@ pub fn push_updates(
...
                 update.qualified_name
             ));
         } else {
             refspecs.push(format!(":{}", update.qualified_name));
         }
     }
-    let result = push_refs(
+    // Please don't inline `push_refs` before 2024-06-01 without talking to ilyagr,
+    // he's working on this code.
```
Comment (yuja) [inline lib/src/git.rs:1294]: ??

Could you add some more context? Is it related to force-with-lease?

**Review by yuja — APPROVED** [review]
Nice

File: lib/src/git.rs (hunk 2 above)
Comment (ilyagr) [reply lib/src/git.rs:1294]: Yes. If this really bothers you, I can online the function right now, but I likely want to extract most of push_refs as a separate function.

I'll link to a draft PR in a bit.

Comment (ilyagr) [reply lib/src/git.rs:1294]: See #3522

Comment (ilyagr) [reply lib/src/git.rs:1294]: I might also just remove the comment and merge this. 
This is not particularly important.

Comment (yuja) [reply lib/src/git.rs:1294]: I'm not against leaving `push_refs()` as a separate function. I just mean the comment is unclear (e.g. why it's 2024-06-01, etc.) Something like `TODO(ilyagr): this function will be reused to implement force-with-lease, so don't inline` will help.

File: lib/src/git.rs (hunk 1 above)
Comment (ilyagr) [reply lib/src/git.rs:1286]: `new_target.hex()` returns a String, and this function needs `&[u8]`. I supposed we could implement `Into<git2::Oid>` for CommitId. I might keep it as is for now.

File: lib/src/git.rs (hunk 2 above)
Comment (ilyagr) [reply lib/src/git.rs:1294]: I combined out two comments.

File: lib/src/git.rs (hunk 1 above)
Comment (yuja) [reply lib/src/git.rs:1286]: I mean `git2::Oid` can be removed at all. It's just there because `Display` is implemented as a hex formatting function.

Comment (ilyagr) [reply lib/src/git.rs:1286]: Ah, of course. It's being converted to a string.

**Pull request merged**